core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -4461,6 +4461,51 @@ public static int time(long timestampMillis, String timeZone) {
         / (1000L * 1000L)); // milli > micro > nano
   }
 
+  /** SQL {@code TO_TIMESTAMP_LTZ(date)} function
+  * for a date values. */
+  public static long toTimestampLtzDate(int days) {
+    return timestamp(days);
```

Comment:
having overloaded `timestamp(int)` and `timestamp(long)` functions with different semantics is error-prone

Reply:
Yeah I just was not sure how else to handle the fact that you can provide the function a `DATE` which gets converted to an `int` or you can provide the function with a numeric value (which could be a long).